src/docs/arch-transaction.dox

```diff
@@ -1,8 +1,202 @@
+
+To read a key, WiredTiger first traverses all the updates of that key still in memory until a
+visible update is found. The in-memory updates in WiredTiger are organized as a singly linked
+list with the newest update at the head, called the update chain. If no value is visible on the
+update chain, it checks the version on the disk image, which is the version that was chosen to
+be written to disk in the last reconciliation. If it is still invisible, WiredTiger will search
+the history store to check if there is a version visible to the reader there.
+
+The repeated read guarantee under snapshot isolation may break in one case if the timestamps
+committed to the updates are out of order, e.g,
+
+`U@20 -> U@30 -> U@15`
+
+In the above example, reading with timestamp 15 doesn't guarantee to return the third update.
+In some cases, users may read the second update U@30 if it is moved to the history store.
+
+@subsection Durability
+
+WiredTiger transactions support commit level durability and checkpoint level durability. An
+operation is commit level durable if logging is enabled on the table (@ref arch-logging). After
+it has been successfully committed, the operation is guaranteed to survive restart. An operation
+will only survive across restart under checkpoint durability if it is included in the last
+successful checkpoint.
+
+@section Prepared Transactions
+
+WiredTiger introduces prepared transactions to meet the needs of implementing distributed
+transactions through two-phase commit. Prepared transactions only work under snapshot isolation.
+
+Instead of just having the beginning, operating, and rollback or commit phase, it has a
+prepared phase before the rollback or commit phase. After prepare is called, WiredTiger
+releases the transaction's snapshot and prohibits any more read or write operations on the
+transaction.
+
+By introducing the prepared stage, a two-phase distributed transaction algorithm can rely on
+the prepared state to reach consensus among all the nodes for committing.
+
+Along with the prepared phase, WiredTiger introduces the prepared timestamp and durable
+timestamp. They are to prevent the slow prepared transactions blocking the movement of the
+global stable timestamp, which may cause excessive amount of data being pinned in memory. The
+stable timestamp is allowed to move beyond the prepared timestamp and at the commit time, the
+prepared transaction can then be committed after the current stable timestamp with a larger
+durable timestamp.
+
+The visibility of the prepared transaction is also special when in the prepared state. Since in
+the prepared state, the transaction has released its snapshot, it should be visible to the
+transactions starting after that based on the normal visibility rule. However, the prepared
+transaction has not been committed and cannot be visible yet. In this situation, WiredTiger
+either ignores the prepared update and tries to read the older updates if it is configured to
```

Comment:
I would reword this section to be:
In this situation WiredTiger will return a WT_PREPARE_CONFLICT to indicate to the caller to retry later, or if configured WiredTiger will ignore the prepared update and read older updates.

WiredTiger either ignores the prepared update and tries to read older updates if it is configured to